include/revng/Model/Type/Type.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
```

Comment:
Maybe rename this to `include/revng/Model/AllTypes.h`?
`Type.h` should be for the `model::Type` class.